dlf/plugins/basket/class.tx_dlf_basket.php

```diff
@@ -0,0 +1,962 @@
+     */
+    public function printDocument() {
+
+        $pdfUrl = $this->conf['pdfprint'];
+
+        foreach ($this->piVars['selected'] as $docId => $docValue) {
+
+            if ($docValue['id']) {
+
+                $docData = $this->getDocumentData($docValue['id'], $docValue);
+
+                $pdfUrl .= $docData['urlParams'].$this->conf['pdfparamseparator'];
+
+                $numberOfPages += $docData['pageNums'];
+
+            }
+
+        }
+
+        // get printer data
+        $printerId = $this->piVars['print_action'];
+
+        // get id from db and send selected doc downloadlink
+        $resultPrinter = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
+            '*',
+            'tx_dlf_printer',
+            'tx_dlf_printer.uid="'.intval($printerId).'"'.tx_dlf_helper::whereClause('tx_dlf_basket'),
+            '',
+            '',
+            '1'
+        );
+
+        $printerData = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($resultPrinter);
+
+        // printer is selected
+        if ($printerData) {
+
+            $pdfUrl = $printerData['print'];
+
+            $filename = 'Document_';
+
+            $numberOfPages = 0;
+
+            foreach ($this->piVars['selected'] as $docId => $docValue) {
+
+                if ($docValue['id']) {
+
+                    $filename .= $docValue['id'].'_';
+
+                    $explodeId = explode("_", $docId);
+
+                    $docData = $this->getDocumentData($explodeId[0], $docValue);
+
+                    $pdfUrl .= $docData['urlParams'].$this->conf['pdfparamseparator'];
+
+                    $numberOfPages += $docData['pageNums'];
+
+                }
+
+            }
+
+            $pdfUrl = trim($pdfUrl, '*');
+
+        }
+
+        // protocol
+        $insertArray = array(
+            'pid' => $this->conf['pages'],
+            'file_name' => $pdfUrl,
+            'count_pages' => $numberOfPages,
+            'crdate' => time(),
+        );
+
+        if ($GLOBALS["TSFE"]->loginUser) {
+
+            // internal user
+            $insertArray['user_id'] = $GLOBALS["TSFE"]->fe_user->user['uid'];
+
+            $insertArray['name'] = $GLOBALS["TSFE"]->fe_user->user['username'];
+
+            $insertArray['label'] = 'Print: '.$printerData['label'];
+
+        } else {
+
+            // external user
+            $insertArray['user_id'] = 0;
+
+            $insertArray['name'] = 'n/a';
+
+            $insertArray['label'] = 'Print: '.$printerData['label'];
+
+        }
+
+        // add action to protocol
+        $GLOBALS['TYPO3_DB']->exec_INSERTquery('tx_dlf_actionlog', $insertArray);
+
+        header('Location: '.$pdfUrl);
+
+        ob_end_flush();
+
+        exit;
+
+    }
+
+}
+
+if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dlf/plugins/basket/class.tx_dlf_basket.php'])	{
+    include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/dlf/plugins/basket/class.tx_dlf_basket.php']);
+}
```

Comment:
Please remove XCLASS definition as this is no longer supported in TYPO3. (Just delete those 3 lines.)